lib/Frontend/ArgsToFrontendOptionsConverter.cpp

```diff
@@ -269,8 +269,17 @@ bool ArgsToFrontendOptionsConverter::convert(
   if (const Arg *A = Args.getLastArg(OPT_module_link_name))
     Opts.ModuleLinkName = A->getValue();
 
-  if (const Arg *A = Args.getLastArg(OPT_package_name))
-    Opts.PackageName = A->getValue();
+  if (const Arg *A = Args.getLastArg(OPT_package_name)) {
+    auto pkgName = A->getValue();
+    if (!Lexer::isIdentifier(pkgName))
+      Diags.diagnose(SourceLoc(), diag::error_bad_package_name, pkgName);
+    else if (pkgName == STDLIB_NAME)
+      Diags.diagnose(SourceLoc(), diag::error_stdlib_package_name, pkgName);
+    else if (pkgName == Opts.ModuleName)
```

Comment:
What's the thinking behind this restriction? 
I imagine many packages out there in the wild having individual targets whose module name matches the name of the package itself. 

e.g. 
https://github.com/apple/swift-driver/blob/main/Package.swift#L21
https://github.com/jpsim/Yams/blob/main/Package.swift#L24
https://github.com/mapbox/mapbox-maps-ios/blob/main/Package.swift#L25

Reply:
I was actually debating on whether to add this restriction or not. Looks like it'd be better without.